Scripts/Engine/CustomBar.lua

```diff
@@ -0,0 +1,848 @@
+-- @tparam number rot rotation of the bar's sprite (0-360).
+-- @tparam Vec2 scaleBar X,Y Scaling factor for the bar's sprite.
+-- @tparam View.AlignMode alignMode Alignment for the bar.
+-- @tparam View.ScaleMode scaleMode Scaling for the bar.
+-- @tparam Effects.BlendID blendMode Blending modes for the bar.
+-- @tparam string text Text to display for the enemy.
+-- @tparam Vec2 textPos X,Y position of the text.
+-- @tparam Strings.DisplayStringOption textOptions alignment and effects for the text. Default: None. Please note text is automatically aligned to the LEFT
+-- @tparam number textScale Scale factor for the text.
+-- @tparam Color textColor Color of the text.
+-- @tparam bool hideText Whether to hide the text.
+-- @tparam number alphaBlendSpeed Speed of alpha blending for bar visibility (0-255).
+-- @tparam string object Enemy name set in Editor for which to create HP for.
+-- @tparam bool showBar Option to always show the bar whether the enemy is current target or not. Useful for boss health bars.
+-- @tparam bool blink Whether the bar blinks.
+-- @tparam number blinkLimit %Limit below which bar starts blinking (0-1).
+--
+-- @treturn CustomBar The custombar in its hidden state
+CustomBar.CreateEnemyHpBar = function (barName,
+							objectIDbg, spriteIDbg, colorbg, posBG, rotBG, scaleBG, alignModebg, scaleModebg, blendModebg, 
+							objectIDbar, spriteIDbar, colorbar, posBar, rot, scaleBar, alignMode, scaleMode, blendMode,
+							text, textPos, textOptions, textScale, textColor, hideText, alphaBlendSpeed, object, showBar, blink, blinkLimit)
+
+    local dataName	= barName .. "_bar_data"
+	local enemyHP = TEN.Objects.GetMoveableByName(object):GetHP()
+
+
+	CustomBar.Create(barName, enemyHP, enemyHP,
+							objectIDbg, spriteIDbg, colorbg, posBG, rotBG, scaleBG, alignModebg, scaleModebg, blendModebg, 
+							objectIDbar, spriteIDbar, colorbar, posBar, rot, scaleBar, alignMode, scaleMode, blendMode,
+							text, textPos, textOptions, textScale, textColor, hideText, alphaBlendSpeed, blink, blinkLimit)
+	
+	LevelVars.Engine.CustomBars.bars[dataName].showBar			= showBar
+	LevelVars.Engine.CustomBars.bars[dataName].object			= object
+	LevelVars.Engine.CustomBars.bars[dataName].getActionType	= 0
+	LevelVars.Engine.CustomBars.bars[dataName].visible			= true
+	LevelVars.Engine.CustomBars.bars[dataName].fixedInterval	= 1/3
+	LevelVars.Engine.CustomBars.bars[dataName].currentAlpha		= 0
+	LevelVars.Engine.CustomBars.bars[dataName].targetAlpha		= 255
+	
+end
+
+---
+-- Generates health bars for all enemies. A new bar appears whenever Lara aims at an enemy. If the "hide text" option is disabled, the enemy's name (as set in the editor) is displayed. Multiple enemies can share the same name by appending _number to the name in the editor. If adjusting an enemy's max HP, ensure this is done before Lara targets the enemy. To create health bars for specific enemies, use CustomBar.CreateEnemyHpBar, ensuring the bar is created prior to targeting.

```

Comment:
Please use line breaks in long LDoc comments for readability.

Reply:
Done